zarr/core.py

```diff
@@ -165,6 +165,9 @@ def _load_metadata_nosync(self):
             if config is None:
                 self._compressor = None
             else:
+                # temporary workaround for
+                # https://github.com/zarr-developers/numcodecs/issues/78
+                config = dict(config)
```

Comment:
Reverting in PR ( https://github.com/zarr-developers/zarr/pull/361 ) as this was fixed in Numcodecs 0.6.0 with PR ( https://github.com/zarr-developers/numcodecs/pull/79 ). As we now require Numcodecs 0.6.0+ in Zarr, we get the fix and thus no longer need the workaround.